lpac-mbim

```diff
@@ -76,8 +83,8 @@ def handle_type_apdu(func: str, param: str):
 
     if func == "logic_channel_close":
         try:
-            print(f"INFO: Close channel {param}")
-            close_channel(int(param))
+            print(f"INFO: Close channel {CHANNEL_ID}")
+            close_channel(int(CHANNEL_ID))
```

Comment:
I think this is not particularly correct (well it works), but lpac is returning a bogus channel number because you're not giving it the correct channel number in logic_channel_open. You're always returning 1 there while the return value should be the channel number it seems https://github.com/z3ntu/lpac-libqmi-wrapper/blob/main/lpac-libqmi-wrapper.py#L90

Reply:
The original code had your suggestion, but @AndySchroder said that for him was not working. Very strange because I tested it and never had a problem.

I'll check it again when receive new modules, in case i'll revert this change back to your original one

Reply:
I think the old way was always returning 1, but the new way I have changed it to uses the correct channel id.

Reply:
I don't think you can specify the channel number in logic_channel_open. It's not that it is not specified correctly, it isn't an input at all, only an output. Is that correct?

Reply:
As mentioned in https://github.com/stich86/lpac-libmbim-wrapper/pull/6#issuecomment-2132291278 , the `param` variable is used in about 3 different ways, I think that is why this bug was created.

Reply:
The application id is the input to logic_channel_open, not the channel id.

Reply:
Wait, you are saying that

https://github.com/stich86/lpac-libmbim-wrapper/blob/a4a2c7b4b7cf2f01f981608af1f5141895ec5084/lpac-mbim#L79

is wrong and should match

https://github.com/z3ntu/lpac-libqmi-wrapper/blob/main/lpac-libqmi-wrapper.py#L90

?

If that change is made, then lpac will keep track of the channel id correctly and pass it around in `param` instead of the wrapper needed to do it itself? If so, I did not realize that lpac did that and so that is why I figured the `param` variable was not being used properly in all places.

Reply:
> Wait, you are saying that
> 
> https://github.com/stich86/lpac-libmbim-wrapper/blob/a4a2c7b4b7cf2f01f981608af1f5141895ec5084/lpac-mbim#L79
> 
> is wrong and should match
> 
> https://github.com/z3ntu/lpac-libqmi-wrapper/blob/main/lpac-libqmi-wrapper.py#L90
> 
> ?
> 
> If that change is made, then lpac will keep track of the channel id correctly and pass it around in `param` instead of the wrapper needed to do it itself? If so, I did not realize that lpac did that and so that is why I figured the `param` variable was not being used properly in all places.

Yes, I'm saying that. Try it yourself, return `"ecode": 2`, and you should then get this parameter in the close function

I didn't really check the code inside lpac itself, but from my observation the return value from channel_open will be the parameter you get for channel_close.

Reply:
okay, but why do you use the global variable `CHANNEL_ID` in the `send_apdu` function then?

Reply:
It seems as though lpac will only pass the channel id as the parameter for closing the channel, but nothing else?

Reply:
Because there we don't get the channel id again, so we need to persist it ourselves regardless. Imo the channel id should also be provided by lpac as parameter for that function, but they don't ..

Reply:
okay, so is the fix I've done more consistent with the rest of the logic in the wrapper required or should we change it?